python/mcap/mcap/exceptions.py

```diff
@@ -13,3 +18,16 @@ class DecoderNotFoundError(McapError):
 
 class EndOfFile(McapError):
     pass
+
+
+class RecordLengthLimitExceeded(McapError):
+    def __init__(self, opcode: int, length: int, limit: int):
+        opcode_name = f"unknown (opcode {opcode})"
+        try:
+            opcode_name = Opcode(opcode).name
+        except ValueError:
+            # unknown opcode will trigger a ValueError
+            pass
+        super().__init__(
+            f"{opcode_name} record has length {length} that exceeds limit {limit}",
+        )
```

Comment:
Nit: could possibly tighten up the language a bit. E.g. from

`unknown (opcode 51) record has length 4049071657446291766 that exceeds limit 4294967296`

to 

`Unknown record (opcode 51) has length 4049071657446291766, which exceeds limit 4294967296`


```suggestion
        record_name = None
        try:
            record_name = f"{Opcode(opcode).name} record"
        except ValueError:
            # unknown opcode will trigger a ValueError
            record_name = f"Unknown record (opcode {opcode})"
        super().__init__(
            f"{record_name} has length {length} that exceeds limit {limit}",
        )
```